src/components/pages/home_screen/sidebar/Sidebar.tsx

```diff
@@ -0,0 +1,256 @@
+import { useState } from "react";
+import {
+  List,
+  ListItem,
+  ListItemIcon,
+  ListItemText,
+  TextField,
+  Typography,
+  Button,
+  Collapse,
+  InputAdornment,
+  IconButton,
+} from "@mui/material";
+import { Category, Task } from "../types";
+import UpcomingIcon from "@mui/icons-material/Upcoming";
+import TodayIcon from "@mui/icons-material/Today";
+import CalendarMonthIcon from "@mui/icons-material/CalendarMonth";
+import AddIcon from "@mui/icons-material/Add";
+import FiberManualRecordIcon from "@mui/icons-material/FiberManualRecord";
+import SearchIcon from "@mui/icons-material/Search";
+import MenuIcon from "@mui/icons-material/Menu";
+import SettingsIcon from "@mui/icons-material/Settings";
+import LogoutIcon from "@mui/icons-material/Logout";
+import DeleteIcon from "@mui/icons-material/Delete";
+
+import "./_sideBar.scss";
+
+interface SidebarProps {
+  categories: Category[];
+  addCategory: (name: string, color: string) => void;
+  deleteCategory: (categoryId: string) => void;
+  isOpen: boolean;
+  toggleSidebar: () => void;
+  onLogout: () => void;
+  setSelectedCategory: (categoryId: string | null) => void;
+  selectedCategory: string | null;
+  tasks: Task[];
+  onSettingsClick: () => void;
+}
+
+const Sidebar = ({
+  categories,
+  addCategory,
+  deleteCategory,
+  isOpen,
+  toggleSidebar,
+  onLogout,
+  setSelectedCategory,
+  selectedCategory,
+  tasks,
+  onSettingsClick,
+}: SidebarProps) => {
+  const [newListName, setNewListName] = useState("");
+  const [selectedColor, setSelectedColor] = useState("#1976d2");
+  const [isAddingNewList, setIsAddingNewList] = useState(false);
+
+  const handleAddCategory = () => {
+    if (newListName) {
+      addCategory(newListName, selectedColor);
+      setNewListName("");
+      setIsAddingNewList(false);
+    }
```

Comment:
Em nên viết kiểu này nhé. Để action của chính function ra ngoài, và check các điều kiện không pass sẽ return luôn

```suggestion
    if (!newListName) {
       return;
    }
    
    addCategory(newListName, selectedColor);
    setNewListName("");
    setIsAddingNewList(false);
  
```